plugins/browser-plugin-snowplow-ecommerce/src/ua/utils.ts

```diff
@@ -0,0 +1,29 @@
+import { Product, Promotion } from '../types';
+import { EEProduct, EEPromo } from './types';
+
+export function transformUAProductsToSPProducts(products: EEProduct[], currency: string): Product[] {
+  return products.map((product) => {
+    return {
+      currency,
+      /* Id or name is required, but we in SP-plugin we require id to be present */
+      id: product.id! || product.name!,
```

Comment:
If we're sure one of the values is defined we shouldn't need the `!`?

Reply:
The `product` in this case comes from the type `EEProduct` which has both `id` and `name` as optional. 

The type inference of the expression in the `id` attribute results in possible `string | undefined` which is not valid for our _(Snowplow Ecommerce)_ API. 

Instead of creating a type helper chose to be explicit about the inference of "one of the two will be there", using the `!` mark. How would you go about it ?

Reply:
I would probably change the type and define `EEProduct` as `({id: string; name?: string} | {id?: string: name: string}) & {...}` to more closely match its actual type. But I think this is fine, I'm just being pedantic; we know it'll be fine but two `!` on a single line made me uncomfortable. :stuck_out_tongue: